File: libs/common/models/vw-program.ts



export class VwProgram {
    public programId: number = 0;
    public programGovernor: string = '';
    public programNumber: number;
    public programName: string = '';
    public programCode: string = '';
    public programLength: string = '';
    public programCredentialId: number;
    public programTypeId: number;
    public programStatus: string = '';
    public providerId: number;
    public businessStartDate: Date;
    public businessEndDate: Date;

    constructor(init? : Partial<VwProgram>) {
        
        Object.assign(this, init);
    }
}

File: apps/institution-api/Models/EntityFramework/VwProgram.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

#nullable disable

namespace ProviderApi.Models.EntityFramework
{
    public partial class VwProgram
    {
        public int ProgramId { get; set; }
        public string ProgramGovernor { get; set; }
        public int? ProgramNumber { get; set; }
        public string ProgramName { get; set; }
        public string ProgramCode { get; set; }
        public string ProgramLength { get; set; }
        public int? ProgramCredentialId { get; set; }
        public int? ProgramTypeId { get; set; }
        public string ProgramStatus { get; set; }
        public int? ProviderId { get; set; }
        public DateTime? BusinessStartDate { get; set; }
        public DateTime? BusinessEndDate { get; set; }

        [NotMapped]
        public int? ProgramEmploymentStat { get; set; }
        [NotMapped]
        public int? ProgramMedianIncome { get; set; }
  }
}
